datafusion/core/src/execution/context.rs

```diff
@@ -1055,12 +1050,16 @@ impl SessionConfig {
         self.set(key, ScalarValue::Boolean(Some(value)))
     }
 
+    /// Set a u32 configuration option
+    pub fn set_u32(self, key: &str, value: u32) -> Self {
+        self.set(key, ScalarValue::UInt32(Some(value)))
+    }
+
     /// Customize batch size
-    pub fn with_batch_size(mut self, n: usize) -> Self {
+    pub fn with_batch_size(self, n: usize) -> Self {
         // batch size must be greater than zero
         assert!(n > 0);
-        self.batch_size = n;
-        self
+        self.set_u32(OPT_BATCH_SIZE, n as u32)
```

Comment:
Why not use `u64` as batch size (why are we storing as a u32 and then casting back and forth to `usize`)?